Initial keyboard navigation

etokareva opened the pull request.

etokareva committed:
shortcut, navigation init

etokareva committed:
navigation model filling

etokareva committed:
fix getting correct charCodes

etokareva committed:
highlight td

etokareva committed:
names fixes

vkorolev commented on src/core/navigation/navigation.js:
```diff
@@ -0,0 +1,44 @@
+					}
+				}),
+				goUp: new Command({
+					shortcut: 'shift+tab|up',
```
shift + tab is opposite to tab. So  i think shift + tab should be in goLeft command

etokareva committed:
shorcut for left\right move

etokareva committed:
Merge branch 'master' of https://github.com/qgrid/ng into keyboard-navigation

Conflicts:
	src/themes/default/assets/body.scss
	src/themes/material/assets/body.scss

vkorolev commented on src/core/infrastructure/shortcut.js:
```diff
@@ -0,0 +1,57 @@
+// import EventListener from './event.listener'; - TODO:: use instead of document
+
+export default function Shortcut(document) {
```
Can you please briefly describe shortcut contract in docs? I mean, describe delimiters ( | +) and a list of encoded values

etokareva committed:
import mixins for default theme;
add color variable based on material indigo theme : https://material.io/guidelines/style/color.html#color-color-palette
https://material.angularjs.org/1.1.0/demo/colors

etokareva committed:
refactoring

etokareva committed:
shortcut docs

etokareva committed:
shortcut docs

etokareva committed:
Navigation thoughts

etokareva committed:
shortcut refactoring;
 navigation styles improvements

etokareva committed:
focused cell styles

vkorolev commented on src/themes/default/assets/body.scss:
```diff
@@ -4,62 +4,70 @@
+  table {
+    tbody {
+      > tr {
+        height: $row-height;
```
Remove spaces please :)

etokareva replied:
this is auto-refactoring)
Have we got tabs?

klumba12 commented on docs/navigation.md:
```diff
@@ -1,12 +1,48 @@
+| Grid          | `control+A` to select the entire grid                                                               |
+
+### `Shift+F8`
+Adds additional discontinuous items to the selection set. If the user has already selected a column then shift+f8 will add additional columns to the set. Similar behavior for rows and cells.
```
terrific job!

klumba12 commented on src/ng/components/body/td.core.js:
```diff
@@ -23,6 +23,14 @@ class TdCore extends Directive(TD_CORE_NAME, {view: `^^${VIEW_CORE_NAME}`}) {
 		}
 
 		this.mode('init');
+		this.view.model.navigationChanged.on(e => {
```
need to think how to move it from td.core...

klumba12 commented on src/themes/default/assets/body.scss:
```diff
@@ -4,62 +4,70 @@
+        td {
+          color: $dark;
+          font-size: 13px;
+          border: {
```
don't need borders i guess?

etokareva replied:
I think, that they can be useful for future if we will change cells' styles

etokareva committed:
navigation improvements

etokareva committed:
Merge branches 'keyboard-navigation' and 'master' of https://github.com/qgrid/ng into keyboard-navigation

# Conflicts:
#	src/ng/components/grid/grid.js
#	src/ng/services/path.find.js

etokareva committed:
formatting
